apps/meteor/app/livechat/server/lib/QueueManager.ts

```diff
@@ -170,6 +182,12 @@ export class QueueManager {
 
 			void callbacks.run('livechat.chatQueued', room);
 
+			if (defaultAgent) {
+				logger.debug(`Setting default agent for inquiry ${inquiry._id} to ${defaultAgent.username}`);
+				await LivechatInquiry.setDefaultAgentById(inquiry._id, defaultAgent);
+				await LivechatVisitors.updateOne({ _id: inquiry.v._id }, { $set: { agentId: defaultAgent.agentId } });
```

Comment:
Maybe move this to a (proper-named) model method? 🤔

Reply:
btw, it doesn't look `agentId` is a valid property of the visitor object:

https://github.com/RocketChat/Rocket.Chat/blob/b6c380d74e4369ab1f9ec7663a9115c83b301e6c/packages/core-typings/src/ILivechatVisitor.ts#L28-L55